plugins/amdgpu/amdgpu_plugin.c

```diff
@@ -390,53 +399,75 @@ int amdgpu_plugin_dump_file(int fd, int id)
 		    KFD_IOC_ALLOC_MEM_FLAGS_VRAM ||
 		    (bo_bucket_ptr)[i].bo_alloc_flags &
 		    KFD_IOC_ALLOC_MEM_FLAGS_GTT) {
+			if ((e->bo_info_test[i])->bo_alloc_flags &
+			    KFD_IOC_ALLOC_MEM_FLAGS_PUBLIC) {
+				pr_info("amdgpu_plugin: large bar read possible\n");
+				addr = mmap(NULL,
+					    (bo_bucket_ptr)[i].bo_size,
+					    PROT_READ,
+					    MAP_SHARED,
+					    drm_fd,	/* mapping on local gpu for prototype */
+					    (bo_bucket_ptr)[i].bo_offset);
+				if (addr == MAP_FAILED) {
+					pr_perror("amdgpu_plugin: mmap failed");
+					fd = -EBADFD;
+					close(drm_fd);
+					goto failed;
+				}
 
-			pr_info("Now try reading BO contents with /proc/pid/mem\n");
-			if (asprintf (&fname, PROCPIDMEM, helper_args.task_pid) < 0) {
-				pr_perror("failed in asprintf, %s", fname);
-				ret = -1;
-				goto failed;
-			}
+				/* direct memcpy is possible on large bars */
+				memcpy((e->bo_info_test[i])->bo_rawdata.data,
+				       addr, bo_bucket_ptr[i].bo_size);
+				munmap(addr, bo_bucket_ptr[i].bo_size);
 
-			mem_fd = open (fname, O_RDONLY);
-			if (mem_fd < 0) {
-				pr_perror("Can't open %s for pid %d", fname, helper_args.task_pid);
-				free (fname);
-				ret = -1;
-				goto failed;
-			}
+			} else {
+				pr_info("Now try reading BO contents with /proc/pid/mem\n");
```

Comment:
"Now try" could be dropped because it doesn't provide the user with any additional information.
```C
pr_info("Reading BO contents with /proc/pid/mem\n");
```

Reply:
ACK, Ditto, as above comment